app/src/main/java/com/github/mik629/android/fundamentals/ui/moviedetails/FragmentMovieDetails.kt

```diff
@@ -1,69 +1,56 @@
 package com.github.mik629.android.fundamentals.ui.moviedetails
 
 import android.os.Bundle
-import android.view.LayoutInflater
 import android.view.View
-import android.view.ViewGroup
 import androidx.fragment.app.Fragment
-import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
+import by.kirich1409.viewbindingdelegate.viewBinding
 import com.github.mik629.android.fundamentals.BuildConfig
-import com.github.mik629.android.fundamentals.GlideApp
 import com.github.mik629.android.fundamentals.R
 import com.github.mik629.android.fundamentals.databinding.FragmentMovieDetailsBinding
-import com.github.mik629.android.fundamentals.domain.model.MovieItem
+import com.github.mik629.android.fundamentals.di.AppModule
+import com.github.mik629.android.fundamentals.di.buildGlideRequest
+import com.github.mik629.android.fundamentals.domain.model.Movie
 import com.github.mik629.android.fundamentals.ui.global.ActorItemAdapter
+import com.github.mik629.android.fundamentals.ui.utils.setRating
 
-class FragmentMovieDetails : Fragment() {
-    private lateinit var binding: FragmentMovieDetailsBinding
+class FragmentMovieDetails : Fragment(R.layout.fragment_movie_details) {
+    private val binding by viewBinding(FragmentMovieDetailsBinding::bind)
 
     private val glideRequest by lazy {
-        GlideApp.with(this)
-            .asDrawable()
-            .thumbnail(0.1f)
-            .placeholder(R.drawable.ic_image_loading)
-            .fallback(R.drawable.ic_broken_image)
-            .transition(DrawableTransitionOptions.withCrossFade())
-            .fitCenter()
+        buildGlideRequest(this)
     }
 
-    private var movieItem: MovieItem? = null
+    private val viewModel by lazy {
+        AppModule.instance.provideViewModel(requireContext())
+    }
+
+    private var movie: Movie? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        movieItem = (savedInstanceState ?: arguments)?.getParcelable(ARG_MOVIE_ITEM)
+        movie = arguments?.getString(ARG_MOVIE_ID)
```

Comment:
make it async with LiveData

Reply:
You mean with LiveData in ViewModel or local LiveData and ViewModel subscribes to it?